src/Microsoft.NetCore.Analyzers/Core/Security/DoNotUseInsecureDeserializerWithoutBinderBase.cs

```diff
@@ -0,0 +1,196 @@
+                                                this.DeserializerTypeMetadataName,
+                                                true /* isNewInstanceFlagged */,
+                                                this.SerializationBinderPropertyMetadataName,
+                                                true /* isNullPropertyFlagged */,
+                                                cachedDeserializationMethodNames);
+                                        if (!dfaResult.Any())
+                                        {
+                                            continue;
+                                        }
+
+                                        if (allResults == null)
+                                        {
+                                            allResults = new Dictionary<OperationMethodKey, PropertySetAbstractValue>();
+                                        }
+
+                                        foreach (KeyValuePair<OperationMethodKey, PropertySetAbstractValue> kvp in dfaResult)
+                                        {
+                                            allResults.Add(kvp.Key, kvp.Value);
+                                        }
+                                    }
+
+                                    if (allResults == null)
+                                    {
+                                        return;
+                                    }
+
+                                    foreach (OperationMethodKey operationMethodKey in allResults.Keys.OrderBy(OperationMethodKey.Comparer))
+                                    {
+                                        PropertySetAbstractValue abstractValue = allResults[operationMethodKey];
+                                        DiagnosticDescriptor descriptor;
+                                        switch (abstractValue)
+                                        {
+                                            case PropertySetAbstractValue.Flagged:
+                                                descriptor = this.BinderDefinitelyNotSetDescriptor;
+                                                break;
+
+                                            case PropertySetAbstractValue.MaybeFlagged:
+                                                descriptor = this.BinderMaybeNotSetDescriptor;
+                                                break;
+
+                                            default:
+                                                Debug.Assert(false, $"Unhandled abstract value {abstractValue}");
+                                                continue;
+                                        }
+
+                                        operationBlockAnalysisContext.ReportDiagnostic(
+                                            Diagnostic.Create(
+                                                descriptor,
+                                                operationMethodKey.Operation.Syntax.GetLocation(),
```

Comment:
>operationMethodKey.Operation.Syntax.GetLocation() [](start = 48, length = 49)

Seems like use are only using the location from the key. Is it possible to declare your hazardous map to be `Dictionary<(Location location, IMethodSymbol method), PropertySetAbstractValue>`.